e2e/ceph_user.go

```diff
@@ -0,0 +1,89 @@
+package e2e
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+// #nosec because of the word `Secret`
+const (
+	// ceph user names
+	keyringRBDProvisionerUsername          = "cephcsi-rbd-provisioner"
+	keyringRBDNodePluginUsername           = "cephcsi-rbd-node"
+	keyringRBDNamespaceProvisionerUsername = "cephcsi-rbd-ns-provisioner"
+	keyringRBDNamespaceNodePluginUsername  = "cephcsi-rbd-ns-node"
+	keyringCephFSProvisionerUsername       = "cephcsi-cephfs-provisioner"
+	keyringCephFSNodePluginUsername        = "cephcsi-cephfs-node"
+	// secret names
+	rbdNodePluginSecretName           = "cephcsi-rbd-node"
+	rbdProvisionerSecretName          = "cephcsi-rbd-provisioner"
+	rbdNamespaceNodePluginSecretName  = "cephcsi-rbd-ns-node"
+	rbdNamespaceProvisionerSecretName = "cephcsi-rbd-ns-provisioner"
+	cephFSNodePluginSecretName        = "cephcsi-cephfs-node"
+	cephFSProvisionerSecretName       = "cephcsi-cephfs-provisioner"
+)
+
+func rbdNodePluginCaps(pool, rbdNamespace string) []string {
+	caps := []string{
```

Comment:
I would love to have a link to the description where the requirements for the capabilities are documented. When e2e fails because of capabilities and these get updated here, there should be a reminder to update the docs too.

Maybe document the required capabilities per function, and then have the document refer to the generated godoc for these functions? https://pkg.go.dev/github.com/ceph/ceph-csi/e2e does not show any documentation for e2e now, not sure what would be missing to generate it...

Reply:
we have documented the required caps at https://github.com/ceph/ceph-csi/blob/devel/docs/capabilities.md added a link for the same in E2E.

Reply:
yes, that is the document I referred to. But as the comment is not added to every function (not needed either), it is easy to forget updating it. the other way around, have the comments for the functions describe the need, and refer to the functions from the document would result in an always up-to-date status. But, that requires the functions to be exposed to godoc...